apps/www/components/Audio/AudioPlayerContainer.tsx

```diff
@@ -0,0 +1,622 @@
+
+  const mediaRef = useRef<HTMLAudioElement>(null)
+  const trackedPlayerItem = useRef<AudioQueueItem>(null)
+  const trackedQueue = useRef<AudioQueueItem[]>(null)
+
+  const [initialized, setInitialized] = useState(false)
+  const [activePlayerItem, setActivePlayerItem] =
+    useState<AudioQueueItem | null>(null)
+  const [shouldAutoPlay, setShouldAutoPlay] = useState(false)
+  const [hasAutoPlayed, setHasAutoPlayed] = useState(false)
+
+  const [isPlaying, setIsPlaying] = useState(false)
+  const [isLoading, setIsLoading] = useState(true)
+  const [isVisible, setIsVisible] = useState(false)
+  const [hasError, setHasError] = useState(false)
+
+  const [currentTime, setCurrentTime] = useState(0)
+  const [duration, setDuration] = useState(0)
+  const [buffered, setBuffered] = useState<TimeRanges>(null)
+  const [playbackRate, setPlaybackRate] = usePlaybackRate(DEFAULT_PLAYBACK_RATE)
+
+  const handleError = (error: Error | string) => {
+    if (typeof error === 'string') {
+      reportError('handle audio-error', error)
+      trackEvent([AUDIO_PLAYER_TRACK_CATEGORY, 'error', error])
+    } else {
+      reportError(
+        'handle audio-error',
+        JSON.stringify(error, Object.getOwnPropertyNames(error), 2),
+      )
+      trackEvent([
+        AUDIO_PLAYER_TRACK_CATEGORY,
+        'error',
+        JSON.stringify(error, Object.getOwnPropertyNames(error), 2),
+      ])
+    }
+  }
+
+  const saveActiveItemProgress = useCallback(
+    async (forcedState?: { currentTime?: number; isPlaying?: boolean }) => {
+      const { mediaId } = activePlayerItem?.document.meta?.audioSource ?? {}
+      if (duration < (forcedState?.currentTime ?? currentTime)) {
+        trackEvent([
+          AUDIO_PLAYER_TRACK_CATEGORY,
+          'updateProgress',
+          'illegalUpdate',
+          `upsert-value was ${
+            forcedState?.currentTime ?? currentTime
+          }, track duration is ${duration}, mediaId is ${mediaId}`,
+        ])
+        return
+      }
+
+      return saveMediaProgress(
+        mediaId,
+        forcedState?.currentTime ?? currentTime,
+        forcedState?.isPlaying ?? isPlaying,
+      )
+    },
+    [activePlayerItem, currentTime, isPlaying, saveMediaProgress, duration],
+  )
+
+  const syncWithNativeApp = (state: AudioPlayerState) => {
+    if (!inNativeApp) return
+    // Don't update the currentTime & duration
+    // if the native player wasn't yet able to load the media
+    // TODO: find out why the track-player sometimes provides currentTime = 0
+    // when state is ready
+    if (
+      ![
+        NativeAudioPlayerState.None,
+        NativeAudioPlayerState.Connecting,
+      ].includes(state.playerState)
+    ) {
+      setDuration(state.duration)
+      setCurrentTime(state.currentTime)
+    }
+    setPlaybackRate(state.playRate)
+    setIsPlaying(
+      state?.isPlaying ||
+        [NativeAudioPlayerState.Playing, NativeAudioPlayerState.Ready].includes(
```

Comment:
Double check if these really are the only states that should resolve to `isPlaying=true`